File: Laboratorio3/biseccion.py
# -*- coding: cp1252 -*-
from math import *
def biseccion(a, b, tol=1.0e-8):
    i =0
    if a > b:
        raise ValueError("Intervalo mal definido")
    if f(a) * f(b) >= 0.0:
        raise ValueError("La función debe cambiar de signo en el intervalo")
    if tol <= 0:
        raise ValueError("La cota de error debe ser un número positivo")
    x = (a + b) / 2.0
    while True:
        i = i +1
        if b - a < tol:
            return x
        # Utilizamos la función signo para evitar errores de precisión
        elif (f(a) * f(x)) > 0:
            a = x
        else:
            b = x
        x = (a + b) / 2.0
        print "el numero de iteraciones es",  i

def f(x): return e**(-x**2) - cos(2*x) - 1
a = 0
b = 3
biseccion(a, b)

File: Laboratorio2/biseccion.m
function [ X ] = Biseccion(f,n,a,b,e)
#Metodo de Biseccion
#f: funcion 
#n: numero de iteracion
#a y b: intervalo [a,b]
#e: error permitido
a(1) = a;
b(1) = b;
for i = 1:n
    m(i) = (a(i) + b(i))/2;
    x = a(i);
    Fa = eval(f);
    x = m(i);
    Fb = eval(f);
    if(Fa*Fb) < 0
        a(i+1) = a(i);
        b(i+1) = m(i);
    else
        a(i+1) = m(i);
        b(i+1) = b(i);
    end

    X(i,1) = i;
    X(i,2) = a(i);
    X(i,3) = b(i);
    X(i,4) = m(i);

    if i > 1
        Ea = abs((m(i) - m(i-1))/m(i))*100;
        X(i,5) = Ea;
        if Ea < e
            break;
        end
    end
    
end
end

